fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/FixedDepositAccount.java

```diff
@@ -83,6 +83,21 @@ public class FixedDepositAccount extends SavingsAccount {
     @Transient
     private ConfigurationDomainService configurationDomainService;
 
+    public void setConfigurationDomainService(ConfigurationDomainService configurationDomainService) {
+        this.configurationDomainService = configurationDomainService;
+    }
+
+    /**
+     * Returns an appropriate ExternalId based on configuration settings. If auto-generation is enabled and the provided
+     * externalId is empty, generate a new one.
+     */
+    public ExternalId getExternalId(ExternalId externalId) {
```

Comment:
No need for this, a simple IF-ELSE would be enough which either generate a new one or return empty.